BibBuddy/app/src/main/java/de/bibbuddy/NotesFragment.java

```diff
@@ -96,61 +98,79 @@ public void onPrepareOptionsMenu(Menu menu) {
   public boolean onOptionsItemSelected(MenuItem item) {
     long id = item.getItemId();
     if (id == R.id.menu_note_list_delete) {
-      handleDeleteNote();
+      handleDeleteNote(adapter.getSelectedNoteItems());
     } else if (id == R.id.menu_note_list_help) {
       handleHelpNotesFragment();
     } else if (id == R.id.menu_imprint) {
-      ((MainActivity) getActivity()).openImprint();
+      ((MainActivity) requireActivity()).openImprint();
     }
 
     return super.onOptionsItemSelected(item);
   }
 
-  private void handleDeleteNote() {
+  private void handleDeleteNote(List<NoteItem> itemsToDelete) {
     AlertDialog.Builder alertDeleteBookNote = new AlertDialog.Builder(requireActivity());
     alertDeleteBookNote.setCancelable(false);
+    alertDeleteBookNote.setTitle(R.string.delete_notes);
 
     if (adapter.getSelectedNoteItems().size() > 1) {
-      alertDeleteBookNote.setTitle(R.string.delete_notes);
       alertDeleteBookNote.setMessage(
-          getString(R.string.delete_notes_message) + " " + getString(R.string.delete_warning));
+          getString(R.string.delete_notes_message)
+              + convertNoteListToString(itemsToDelete)
+              + getString(R.string.finally_delete) + " "
+              + getString(R.string.delete_warning));
     } else {
-      alertDeleteBookNote.setTitle(R.string.delete_note);
       alertDeleteBookNote.setMessage(
-          getString(R.string.delete_note_message) + " " + getString(R.string.delete_warning));
+          getString(R.string.delete_note_message)
+              + convertNoteListToString(itemsToDelete)
+              + getString(R.string.finally_delete) + " "
+              + getString(R.string.delete_warning));
```

Comment:
Maybe you could create a separate function for this because above is also the same code in parts.

Reply:
Can not be sensibly decomposed here.